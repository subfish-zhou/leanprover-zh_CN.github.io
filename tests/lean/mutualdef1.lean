--

mutual

def f (x : Nat) : Nat → Nat
| 0   => 1
| x+1 => g x

theorem g (x : Nat) : Nat → Nat -- cannot mix theorems and definitions
| 0   => 2
| x+1 => f x

end

mutual

def f  : Nat → Nat
| 0   => 1
| x+1 => g x

example : Nat := -- cannot mix examples and definitions
g 10

end

mutual

def f (x : Nat) : Nat → Nat
| 0   => 1
| x+1 => g x

unsafe def g (x : Nat) : Nat → Nat -- cannot mix safe and unsafe definitions
| 0   => 2
| x+1 => f x

end
