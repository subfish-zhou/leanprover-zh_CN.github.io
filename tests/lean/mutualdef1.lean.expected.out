mutualdef1.lean:9:0-11:12: error: invalid mutually recursive definitions, cannot mix theorems and definitions
mutualdef1.lean:21:0-22:4: error: invalid mutually recursive definitions, cannot mix examples and definitions
mutualdef1.lean:32:7-34:12: error: invalid mutually recursive definitions, cannot mix unsafe and safe definitions
